### ISSUE TITLE
shakespeare >= 2.0.2 fails to install in OS X 10.6.8, Haskell-platform 2014.2.0.0; cabal-install 1.22.4.0

### ISSUE THREAD

**Issue by MASantos** [yesodweb/shakespeare#156]
(I haven't found any other place for reporting this installation problem. Simply delete if it isn't the right one.)

Doing a cabal install yesod-bin or cabal install yesod fails because it fails to build shakespeare.

I tried with all versions of shapeare >= 2.0, as required by yesod-bin. Below are all outputs.

My environment: 
* OS X 10.6.8, Xcode 3.2
* Haskell-platform 2014.2.0.0
* ghc-7.8.3 
* cabal-install 1.22.4.0 using version 1.22.3.0 of the Cabal library.

---
cabal: Error: some packages failed to install:
shakespeare-2.0.5 failed during the building phase. The exception was:
ExitFailure 1
yesod-bin-1.4.9.2 depends on shakespeare-2.0.5 which failed to install.
msantos@MBP-2[16:37]:~/PRG/HASKELL$cabal install shakespeare-2.0.4.1
Resolving dependencies...
Downloading shakespeare-2.0.4.1...
Configuring shakespeare-2.0.4.1...
Building shakespeare-2.0.4.1...
Failed to install shakespeare-2.0.4.1
Build log ( /Users/msantos/.cabal/logs/shakespeare-2.0.4.1.log ):
Configuring shakespeare-2.0.4.1...
Building shakespeare-2.0.4.1...
Preprocessing library shakespeare-2.0.4.1...
[ 1 of 17] Compiling Text.MkSizeType  ( Text/MkSizeType.hs, dist/build/Text/MkSizeType.o )
[ 2 of 17] Compiling Text.IndentToBrace ( Text/IndentToBrace.hs, dist/build/Text/IndentToBrace.o )
[ 3 of 17] Compiling Text.Shakespeare.Base ( Text/Shakespeare/Base.hs, dist/build/Text/Shakespeare/Base.o )
[ 4 of 17] Compiling Text.Hamlet.Parse ( Text/Hamlet/Parse.hs, dist/build/Text/Hamlet/Parse.o )
[ 5 of 17] Compiling Text.Hamlet      ( Text/Hamlet.hs, dist/build/Text/Hamlet.o )

Text/Hamlet.hs:403:1: Warning: Defined but not used: ‘varName’

Text/Hamlet.hs:407:1: Warning: Defined but not used: ‘strToExp’

Text/Hamlet.hs:440:16: Warning: Defined but not used: ‘html’

Text/Hamlet.hs:441:14: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:442:19: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:443:16: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:444:20: Warning: Defined but not used: ‘msg_url’

Text/Hamlet.hs:445:14: Warning: Defined but not used: ‘msg’

Text/Hamlet.hs:489:43: Warning:
    This binding for ‘c’ shadows the existing binding
      bound at Text/Hamlet.hs:485:13
[ 6 of 17] Compiling Text.Hamlet.RT   ( Text/Hamlet/RT.hs, dist/build/Text/Hamlet/RT.o )

Text/Hamlet/RT.hs:78:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:78:26: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:78:45: Warning: Defined but not used: ‘docs’

Text/Hamlet/RT.hs:85:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:85:25: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:85:44: Warning: Defined but not used: ‘jdocs’

Text/Hamlet/RT.hs:85:50: Warning: Defined but not used: ‘ndocs’

Text/Hamlet/RT.hs:115:37: Warning:
    This binding for ‘x’ shadows the existing binding
      bound at Text/Hamlet/RT.hs:108:13
[ 7 of 17] Compiling Text.Shakespeare ( Text/Shakespeare.hs, dist/build/Text/Shakespeare.o )
[ 8 of 17] Compiling Text.Shakespeare.Text ( Text/Shakespeare/Text.hs, dist/build/Text/Shakespeare/Text.o )
[ 9 of 17] Compiling Text.Julius      ( Text/Julius.hs, dist/build/Text/Julius.o )

Text/Julius.hs:90:63: Warning:
    In the use of ‘fromValue’ (imported from Data.Aeson.Encode):
    Deprecated: "Use 'encodeToTextBuilder' instead"
[10 of 17] Compiling Text.Roy         ( Text/Roy.hs, dist/build/Text/Roy.o )
[11 of 17] Compiling Text.Coffee      ( Text/Coffee.hs, dist/build/Text/Coffee.o )
[12 of 17] Compiling Text.Css         ( Text/Css.hs, dist/build/Text/Css.o )

Text/Css.hs:349:40: Warning: Defined but not used: ‘subblocks’

Text/Css.hs:367:5: Warning: Defined but not used: ‘subGo’
[13 of 17] Compiling Text.CssCommon   ( Text/CssCommon.hs, dist/build/Text/CssCommon.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package text-1.1.0.0 ... linking ... done.
Loading package system-filepath-0.4.13.4 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package system-fileio-0.3.16.3 ... linking ... done.
Loading package filepath-1.3.0.2 ... linking ... done.
Loading package directory-1.2.1.0 ... linking ... done.
Loading package process-1.2.0.0 ... linking ... done.
Loading package transformers-0.3.0.0 ... linking ... done.
Loading package mtl-2.1.3.1 ... linking ... done.
Loading package parsec-3.1.5 ... <command line>: can't load .so/.DLL for: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib (dlopen(/Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib, 9): Library not loaded: @rpath/libHStext-1.1.0.0-ghc7.8.3.dylib
  Referenced from: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
  Reason: image not found)
cabal: Error: some packages failed to install:
shakespeare-2.0.4.1 failed during the building phase. The exception was:
ExitFailure 1
msantos@MBP-2[16:50]:~/PRG/HASKELL$cabal install shakespeare-2.0.4
Resolving dependencies...
Configuring shakespeare-2.0.4...
Building shakespeare-2.0.4...
Failed to install shakespeare-2.0.4
Build log ( /Users/msantos/.cabal/logs/shakespeare-2.0.4.log ):
Configuring shakespeare-2.0.4...
Building shakespeare-2.0.4...
Preprocessing library shakespeare-2.0.4...
[ 1 of 17] Compiling Text.MkSizeType  ( Text/MkSizeType.hs, dist/build/Text/MkSizeType.o )
[ 2 of 17] Compiling Text.IndentToBrace ( Text/IndentToBrace.hs, dist/build/Text/IndentToBrace.o )
[ 3 of 17] Compiling Text.Shakespeare.Base ( Text/Shakespeare/Base.hs, dist/build/Text/Shakespeare/Base.o )
[ 4 of 17] Compiling Text.Hamlet.Parse ( Text/Hamlet/Parse.hs, dist/build/Text/Hamlet/Parse.o )
[ 5 of 17] Compiling Text.Hamlet      ( Text/Hamlet.hs, dist/build/Text/Hamlet.o )

Text/Hamlet.hs:402:1: Warning: Defined but not used: ‘varName’

Text/Hamlet.hs:406:1: Warning: Defined but not used: ‘strToExp’

Text/Hamlet.hs:439:16: Warning: Defined but not used: ‘html’

Text/Hamlet.hs:440:14: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:441:19: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:442:16: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:443:20: Warning: Defined but not used: ‘msg_url’

Text/Hamlet.hs:444:14: Warning: Defined but not used: ‘msg’

Text/Hamlet.hs:488:43: Warning:
    This binding for ‘c’ shadows the existing binding
      bound at Text/Hamlet.hs:484:13
[ 6 of 17] Compiling Text.Hamlet.RT   ( Text/Hamlet/RT.hs, dist/build/Text/Hamlet/RT.o )

Text/Hamlet/RT.hs:78:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:78:26: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:78:45: Warning: Defined but not used: ‘docs’

Text/Hamlet/RT.hs:85:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:85:25: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:85:44: Warning: Defined but not used: ‘jdocs’

Text/Hamlet/RT.hs:85:50: Warning: Defined but not used: ‘ndocs’

Text/Hamlet/RT.hs:115:37: Warning:
    This binding for ‘x’ shadows the existing binding
      bound at Text/Hamlet/RT.hs:108:13
[ 7 of 17] Compiling Text.Shakespeare ( Text/Shakespeare.hs, dist/build/Text/Shakespeare.o )
[ 8 of 17] Compiling Text.Shakespeare.Text ( Text/Shakespeare/Text.hs, dist/build/Text/Shakespeare/Text.o )
[ 9 of 17] Compiling Text.Julius      ( Text/Julius.hs, dist/build/Text/Julius.o )

Text/Julius.hs:90:63: Warning:
    In the use of ‘fromValue’ (imported from Data.Aeson.Encode):
    Deprecated: "Use 'encodeToTextBuilder' instead"
[10 of 17] Compiling Text.Roy         ( Text/Roy.hs, dist/build/Text/Roy.o )
[11 of 17] Compiling Text.Coffee      ( Text/Coffee.hs, dist/build/Text/Coffee.o )
[12 of 17] Compiling Text.Css         ( Text/Css.hs, dist/build/Text/Css.o )

Text/Css.hs:349:40: Warning: Defined but not used: ‘subblocks’

Text/Css.hs:367:5: Warning: Defined but not used: ‘subGo’
[13 of 17] Compiling Text.CssCommon   ( Text/CssCommon.hs, dist/build/Text/CssCommon.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package text-1.1.0.0 ... linking ... done.
Loading package system-filepath-0.4.13.4 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package system-fileio-0.3.16.3 ... linking ... done.
Loading package filepath-1.3.0.2 ... linking ... done.
Loading package directory-1.2.1.0 ... linking ... done.
Loading package process-1.2.0.0 ... linking ... done.
Loading package transformers-0.3.0.0 ... linking ... done.
Loading package mtl-2.1.3.1 ... linking ... done.
Loading package parsec-3.1.5 ... <command line>: can't load .so/.DLL for: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib (dlopen(/Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib, 9): Library not loaded: @rpath/libHStext-1.1.0.0-ghc7.8.3.dylib
  Referenced from: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
  Reason: image not found)
cabal: Error: some packages failed to install:
shakespeare-2.0.4 failed during the building phase. The exception was:
ExitFailure 1
msantos@MBP-2[16:52]:~/PRG/HASKELL$cabal install shakespeare-2.0.3
Resolving dependencies...
Downloading shakespeare-2.0.3...
Configuring shakespeare-2.0.3...
Building shakespeare-2.0.3...
Failed to install shakespeare-2.0.3
Build log ( /Users/msantos/.cabal/logs/shakespeare-2.0.3.log ):
Configuring shakespeare-2.0.3...
Building shakespeare-2.0.3...
Preprocessing library shakespeare-2.0.3...
[ 1 of 17] Compiling Text.MkSizeType  ( Text/MkSizeType.hs, dist/build/Text/MkSizeType.o )
[ 2 of 17] Compiling Text.IndentToBrace ( Text/IndentToBrace.hs, dist/build/Text/IndentToBrace.o )
[ 3 of 17] Compiling Text.Shakespeare.Base ( Text/Shakespeare/Base.hs, dist/build/Text/Shakespeare/Base.o )
[ 4 of 17] Compiling Text.Hamlet.Parse ( Text/Hamlet/Parse.hs, dist/build/Text/Hamlet/Parse.o )
[ 5 of 17] Compiling Text.Hamlet      ( Text/Hamlet.hs, dist/build/Text/Hamlet.o )

Text/Hamlet.hs:402:1: Warning: Defined but not used: ‘varName’

Text/Hamlet.hs:406:1: Warning: Defined but not used: ‘strToExp’

Text/Hamlet.hs:439:16: Warning: Defined but not used: ‘html’

Text/Hamlet.hs:440:14: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:441:19: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:442:16: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:443:20: Warning: Defined but not used: ‘msg_url’

Text/Hamlet.hs:444:14: Warning: Defined but not used: ‘msg’

Text/Hamlet.hs:488:43: Warning:
    This binding for ‘c’ shadows the existing binding
      bound at Text/Hamlet.hs:484:13
[ 6 of 17] Compiling Text.Hamlet.RT   ( Text/Hamlet/RT.hs, dist/build/Text/Hamlet/RT.o )

Text/Hamlet/RT.hs:78:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:78:26: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:78:45: Warning: Defined but not used: ‘docs’

Text/Hamlet/RT.hs:85:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:85:25: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:85:44: Warning: Defined but not used: ‘jdocs’

Text/Hamlet/RT.hs:85:50: Warning: Defined but not used: ‘ndocs’

Text/Hamlet/RT.hs:115:37: Warning:
    This binding for ‘x’ shadows the existing binding
      bound at Text/Hamlet/RT.hs:108:13
[ 7 of 17] Compiling Text.Shakespeare ( Text/Shakespeare.hs, dist/build/Text/Shakespeare.o )
[ 8 of 17] Compiling Text.Shakespeare.Text ( Text/Shakespeare/Text.hs, dist/build/Text/Shakespeare/Text.o )
[ 9 of 17] Compiling Text.Julius      ( Text/Julius.hs, dist/build/Text/Julius.o )

Text/Julius.hs:90:63: Warning:
    In the use of ‘fromValue’ (imported from Data.Aeson.Encode):
    Deprecated: "Use 'encodeToTextBuilder' instead"
[10 of 17] Compiling Text.Roy         ( Text/Roy.hs, dist/build/Text/Roy.o )
[11 of 17] Compiling Text.Coffee      ( Text/Coffee.hs, dist/build/Text/Coffee.o )
[12 of 17] Compiling Text.Css         ( Text/Css.hs, dist/build/Text/Css.o )

Text/Css.hs:349:40: Warning: Defined but not used: ‘subblocks’

Text/Css.hs:367:5: Warning: Defined but not used: ‘subGo’
[13 of 17] Compiling Text.CssCommon   ( Text/CssCommon.hs, dist/build/Text/CssCommon.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package text-1.1.0.0 ... linking ... done.
Loading package system-filepath-0.4.13.4 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package system-fileio-0.3.16.3 ... linking ... done.
Loading package filepath-1.3.0.2 ... linking ... done.
Loading package directory-1.2.1.0 ... linking ... done.
Loading package process-1.2.0.0 ... linking ... done.
Loading package transformers-0.3.0.0 ... linking ... done.
Loading package mtl-2.1.3.1 ... linking ... done.
Loading package parsec-3.1.5 ... <command line>: can't load .so/.DLL for: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib (dlopen(/Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib, 9): Library not loaded: @rpath/libHStext-1.1.0.0-ghc7.8.3.dylib
  Referenced from: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
  Reason: image not found)
cabal: Error: some packages failed to install:
shakespeare-2.0.3 failed during the building phase. The exception was:
ExitFailure 1
msantos@MBP-2[16:54]:~/PRG/HASKELL$cabal install shakespeare-2.0.2.2
Resolving dependencies...
Downloading shakespeare-2.0.2.2...
Configuring shakespeare-2.0.2.2...
Building shakespeare-2.0.2.2...
Failed to install shakespeare-2.0.2.2
Build log ( /Users/msantos/.cabal/logs/shakespeare-2.0.2.2.log ):
Configuring shakespeare-2.0.2.2...
Building shakespeare-2.0.2.2...
Preprocessing library shakespeare-2.0.2.2...
[ 1 of 17] Compiling Text.MkSizeType  ( Text/MkSizeType.hs, dist/build/Text/MkSizeType.o )
[ 2 of 17] Compiling Text.IndentToBrace ( Text/IndentToBrace.hs, dist/build/Text/IndentToBrace.o )
[ 3 of 17] Compiling Text.Shakespeare.Base ( Text/Shakespeare/Base.hs, dist/build/Text/Shakespeare/Base.o )
[ 4 of 17] Compiling Text.Hamlet.Parse ( Text/Hamlet/Parse.hs, dist/build/Text/Hamlet/Parse.o )
[ 5 of 17] Compiling Text.Hamlet      ( Text/Hamlet.hs, dist/build/Text/Hamlet.o )

Text/Hamlet.hs:402:1: Warning: Defined but not used: ‘varName’

Text/Hamlet.hs:406:1: Warning: Defined but not used: ‘strToExp’

Text/Hamlet.hs:439:16: Warning: Defined but not used: ‘html’

Text/Hamlet.hs:440:14: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:441:19: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:442:16: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:443:20: Warning: Defined but not used: ‘msg_url’

Text/Hamlet.hs:444:14: Warning: Defined but not used: ‘msg’

Text/Hamlet.hs:488:43: Warning:
    This binding for ‘c’ shadows the existing binding
      bound at Text/Hamlet.hs:484:13
[ 6 of 17] Compiling Text.Hamlet.RT   ( Text/Hamlet/RT.hs, dist/build/Text/Hamlet/RT.o )

Text/Hamlet/RT.hs:78:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:78:26: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:78:45: Warning: Defined but not used: ‘docs’

Text/Hamlet/RT.hs:85:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:85:25: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:85:44: Warning: Defined but not used: ‘jdocs’

Text/Hamlet/RT.hs:85:50: Warning: Defined but not used: ‘ndocs’

Text/Hamlet/RT.hs:115:37: Warning:
    This binding for ‘x’ shadows the existing binding
      bound at Text/Hamlet/RT.hs:108:13
[ 7 of 17] Compiling Text.Shakespeare ( Text/Shakespeare.hs, dist/build/Text/Shakespeare.o )
[ 8 of 17] Compiling Text.Shakespeare.Text ( Text/Shakespeare/Text.hs, dist/build/Text/Shakespeare/Text.o )
[ 9 of 17] Compiling Text.Julius      ( Text/Julius.hs, dist/build/Text/Julius.o )

Text/Julius.hs:90:63: Warning:
    In the use of ‘fromValue’ (imported from Data.Aeson.Encode):
    Deprecated: "Use 'encodeToTextBuilder' instead"
[10 of 17] Compiling Text.Roy         ( Text/Roy.hs, dist/build/Text/Roy.o )
[11 of 17] Compiling Text.Coffee      ( Text/Coffee.hs, dist/build/Text/Coffee.o )
[12 of 17] Compiling Text.Css         ( Text/Css.hs, dist/build/Text/Css.o )

Text/Css.hs:349:40: Warning: Defined but not used: ‘subblocks’

Text/Css.hs:367:5: Warning: Defined but not used: ‘subGo’
[13 of 17] Compiling Text.CssCommon   ( Text/CssCommon.hs, dist/build/Text/CssCommon.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package text-1.1.0.0 ... linking ... done.
Loading package system-filepath-0.4.13.4 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package system-fileio-0.3.16.3 ... linking ... done.
Loading package filepath-1.3.0.2 ... linking ... done.
Loading package directory-1.2.1.0 ... linking ... done.
Loading package process-1.2.0.0 ... linking ... done.
Loading package transformers-0.3.0.0 ... linking ... done.
Loading package mtl-2.1.3.1 ... linking ... done.
Loading package parsec-3.1.5 ... <command line>: can't load .so/.DLL for: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib (dlopen(/Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib, 9): Library not loaded: @rpath/libHStext-1.1.0.0-ghc7.8.3.dylib
  Referenced from: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
  Reason: image not found)
cabal: Error: some packages failed to install:
shakespeare-2.0.2.2 failed during the building phase. The exception was:
ExitFailure 1
msantos@MBP-2[16:55]:~/PRG/HASKELL$cabal install shakespeare-2.0.2
Resolving dependencies...
Downloading shakespeare-2.0.2...
Configuring shakespeare-2.0.2...
Building shakespeare-2.0.2...
Failed to install shakespeare-2.0.2
Build log ( /Users/msantos/.cabal/logs/shakespeare-2.0.2.log ):
Configuring shakespeare-2.0.2...
Building shakespeare-2.0.2...
Preprocessing library shakespeare-2.0.2...
[ 1 of 17] Compiling Text.MkSizeType  ( Text/MkSizeType.hs, dist/build/Text/MkSizeType.o )
[ 2 of 17] Compiling Text.IndentToBrace ( Text/IndentToBrace.hs, dist/build/Text/IndentToBrace.o )
[ 3 of 17] Compiling Text.Shakespeare.Base ( Text/Shakespeare/Base.hs, dist/build/Text/Shakespeare/Base.o )
[ 4 of 17] Compiling Text.Hamlet.Parse ( Text/Hamlet/Parse.hs, dist/build/Text/Hamlet/Parse.o )
[ 5 of 17] Compiling Text.Hamlet      ( Text/Hamlet.hs, dist/build/Text/Hamlet.o )

Text/Hamlet.hs:402:1: Warning: Defined but not used: ‘varName’

Text/Hamlet.hs:406:1: Warning: Defined but not used: ‘strToExp’

Text/Hamlet.hs:439:16: Warning: Defined but not used: ‘html’

Text/Hamlet.hs:440:14: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:441:19: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:442:16: Warning: Defined but not used: ‘url’

Text/Hamlet.hs:443:20: Warning: Defined but not used: ‘msg_url’

Text/Hamlet.hs:444:14: Warning: Defined but not used: ‘msg’

Text/Hamlet.hs:488:43: Warning:
    This binding for ‘c’ shadows the existing binding
      bound at Text/Hamlet.hs:484:13
[ 6 of 17] Compiling Text.Hamlet.RT   ( Text/Hamlet/RT.hs, dist/build/Text/Hamlet/RT.o )

Text/Hamlet/RT.hs:78:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:78:26: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:78:45: Warning: Defined but not used: ‘docs’

Text/Hamlet/RT.hs:85:13: Warning: Defined but not used: ‘x’

Text/Hamlet/RT.hs:85:25: Warning: Defined but not used: ‘deref’

Text/Hamlet/RT.hs:85:44: Warning: Defined but not used: ‘jdocs’

Text/Hamlet/RT.hs:85:50: Warning: Defined but not used: ‘ndocs’

Text/Hamlet/RT.hs:115:37: Warning:
    This binding for ‘x’ shadows the existing binding
      bound at Text/Hamlet/RT.hs:108:13
[ 7 of 17] Compiling Text.Shakespeare ( Text/Shakespeare.hs, dist/build/Text/Shakespeare.o )
[ 8 of 17] Compiling Text.Shakespeare.Text ( Text/Shakespeare/Text.hs, dist/build/Text/Shakespeare/Text.o )
[ 9 of 17] Compiling Text.Julius      ( Text/Julius.hs, dist/build/Text/Julius.o )

Text/Julius.hs:90:63: Warning:
    In the use of ‘fromValue’ (imported from Data.Aeson.Encode):
    Deprecated: "Use 'encodeToTextBuilder' instead"
[10 of 17] Compiling Text.Roy         ( Text/Roy.hs, dist/build/Text/Roy.o )
[11 of 17] Compiling Text.Coffee      ( Text/Coffee.hs, dist/build/Text/Coffee.o )
[12 of 17] Compiling Text.Css         ( Text/Css.hs, dist/build/Text/Css.o )

Text/Css.hs:349:40: Warning: Defined but not used: ‘subblocks’

Text/Css.hs:367:5: Warning: Defined but not used: ‘subGo’
[13 of 17] Compiling Text.CssCommon   ( Text/CssCommon.hs, dist/build/Text/CssCommon.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package text-1.1.0.0 ... linking ... done.
Loading package system-filepath-0.4.13.4 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package system-fileio-0.3.16.3 ... linking ... done.
Loading package filepath-1.3.0.2 ... linking ... done.
Loading package directory-1.2.1.0 ... linking ... done.
Loading package process-1.2.0.0 ... linking ... done.
Loading package transformers-0.3.0.0 ... linking ... done.
Loading package mtl-2.1.3.1 ... linking ... done.
Loading package parsec-3.1.5 ... <command line>: can't load .so/.DLL for: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib (dlopen(/Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib, 9): Library not loaded: @rpath/libHStext-1.1.0.0-ghc7.8.3.dylib
  Referenced from: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
  Reason: image not found)
cabal: Error: some packages failed to install:
shakespeare-2.0.2 failed during the building phase. The exception was:
ExitFailure 1

**Comment by snoyberg** [comment]
I'm going to guess this is a Haskell Platform bug, but I can't be certain. I'd recommend installing via GHC for Mac OS X (see [install page](http://www.stackage.org/install#mac-os-x)). Can you try that and let me know if the problem persists?

**Comment by MASantos** [comment]
Thanks. I'll try, but it won't be before the weekend. I just had H-Platform re-installed and I need pandoc, which took me a lot to get cabal build and install it.

**Comment by MASantos** [comment]
Sorry for the late answer, but I needed to use pandoc for a project and couldn't afford the time of removing HP. Finally I got that time. Unfortunately, that install page you pointed me to handles only Mac OSX 10.7 and above, and my system is 10.6.8. 

Could you help me hack a fix for it? For one reason or another I never had the chance to play with yesod, as it never installed properly or at all.

**Comment by snoyberg** [comment]
It appears like the bindists provided on haskell.org simply do not have support for anything before 10.7. Perhaps someone on the mailing list would have advice on how to proceed, but unfortunately I have no access to a Mac system.

You could theoretically just try that bindist and see if it works, but I'd assume they have a good reason for that comment.

**Comment by MASantos** [comment]
:) I already thought on that and gave it a try just in case...the ./configure script quits immediately complaining it "cannot determine the current directory".

I was thinking I'd be possible to go through shakespeare code and make it "HP friendly". 

I already sent an email to the Haskell mailing list.

**Comment by mzero** [comment]
The bindist of Haskell Platform is built with 10.6.8 support, including a build of GHC that is build with 10.6.8 support.

However, this does appear to be a problem with template haskell, and GHC setting up the DYLD environment such that it can find all the dependent libs. I suspect it is bug in GHC, and has nothing to do with either HP or how GHC was built.

The work around is, when you see a message like (re-wrapped here for legibility):

    Loading package parsec-3.1.5 ... <command line>: can't load .so/.DLL for:
        /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
        (dlopen(/Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib, 9):
        Library not loaded: @rpath/libHStext-1.1.0.0-ghc7.8.3.dylib
    Referenced from: /Library/Haskell/ghc-7.8.3-x86_64/lib/parsec-3.1.5/libHSparsec-3.1.5-ghc7.8.3.dylib
    Reason: image not found)

Look at the `Library not loaded:` section. It indicates that `text` can't be loaded. You need to add it's libdir to an export like so (you are adding the location of the`libHStext-1.1.0.0-ghc7.8.3.dylib` file, which is always in a similarly named directory under `/Library/Haskell/ghc-7.8.3-x86_64/lib`):

    export DYLD_FALLBACK_LIBRARY_PATH=/Library/Haskell/ghc-7.8.3-x86_64/lib/text-1.1.0.0:$DYLD_FALLBACK_LIBRARY_PATH

Rebuild, and you may see another to add. You may need to do this multiple times - but eventually DYLD_FALLBACK_LIBRARY_PATH will have enough in it that things can build.

I was able to build shaekespeare-2.0.4.1 on a stock HP 2014.2.0.0 install this way.

**Comment by mzero** [comment]
For the record: The problem doesn't occur on OS X 10.10.

I haven't done the "binary search through OS X releases" dance to find where the problem stops being a problem...

**Comment by MASantos** [comment]
Solved!

The quickstart from the yesod site
http://www.yesodweb.com/page/quickstart
wouldn't work as I said at the beginning of this thread because the
shakespeare package didn't get built.
That's because those steps suggest installing yesod-bin globally and only
afterwards start the sandbox.
Reversing those steps does the work.

More concretely, what I did was
mkdir yesod
cd yesod
wget --no-check-certificate https://www.stackage.org/lts/cabal.config
cabal update
cabal install alex happy (actually I had these already installed)
cabal sandbox init
cabal install network (see below why)
cabal install primitive-0.6 (idem)
cabal install yesod-bin

And at this point shakespeare built fine and yesod got installed. I went then further and did
export PATH=./.cabal-sandbox/bin/:$PATH
in order to get yesod running.

yesod init --bare
cabal install -j --enable-tests --max-backjumps=-1 --reorder-goal (probably not necessary; followed blindly init's suggestion)
cabal install --run-tests
yesod devel

That lead to a minor configuring of FirstWebServer-0.0.0..., which triggered a recompile of a few libraries and a rebuild of the application and finally a working Welcome Yesod page at localhost:3000

More details on the Haskell-cafe mailing list under the thread https://mail.haskell.org/pipermail/haskell-cafe/2015-June/120192.html

Thanks both for your help!
MA
